app/src/main/java/com/cvdevelopers/baseapplication/template_generated_classes/TemplateDataRepository.kt

```diff
@@ -0,0 +1,29 @@
+package com.cvdevelopers.baseapplication.template_generated_classes;
+
+import com.cvdevelopers.baseapplication.baseclasses.BaseDataRepository
+import com.cvdevelopers.baseapplication.utils.SubscriptionManager
+import io.reactivex.Observable
+import io.reactivex.subjects.BehaviorSubject
+import javax.inject.Inject
+
+class TemplateDataRepository @Inject constructor() : BaseDataRepository() {
+
+    private val publishViewModelData = BehaviorSubject.create<Any>()
+
+    override fun initializeSubscriptions(subscriptionManager: SubscriptionManager) {
+        subscriptionManager.apply {
+
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        publishViewModelData.onNext("Hard coded text for Template")
+    }
+
+    fun observeRepositoryData() = publishViewModelData as Observable<Any>
```

Comment:
shouldn't this return TemplateDisplayData instead of `Any`? You have it in the view model to create the model, but why not let the data repository do that?